### PR TITLE
Upgrade Hibernate Search to Alpha9

### PR DESCRIPTION
**Pull request by gsmet**
Also fixes a few issues noticed while upgrading.

### CODE REVIEW

File: extensions/hibernate-search-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/elasticsearch/runtime/HibernateSearchElasticsearchRuntimeConfig.java [hunk 1]
```diff
@@ -9,26 +9,40 @@
...
 import org.hibernate.search.mapper.orm.cfg.HibernateOrmAutomaticIndexingSynchronizationStrategyName;
+import org.hibernate.search.mapper.orm.search.loading.EntityLoadingCacheLookupStrategy;
 
 import io.quarkus.runtime.annotations.ConfigGroup;
 import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
 
-@ConfigRoot(name = "hibernate-search", phase = ConfigPhase.RUN_TIME)
+@ConfigRoot(name = "hibernate-search.elasticsearch", phase = ConfigPhase.RUN_TIME)
```
Comment (machi1990) [inline extensions/hibernate-search-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/elasticsearch/runtime/HibernateSearchElasticsearchRuntimeConfig.java]: Breaking change?

Comment (gsmet) [reply extensions/hibernate-search-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/elasticsearch/runtime/HibernateSearchElasticsearchRuntimeConfig.java]: Not really, in the end, it should be the same as before, except you could define some properties per backend whereas they were just ignored.

Comment (machi1990) [reply extensions/hibernate-search-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/elasticsearch/runtime/HibernateSearchElasticsearchRuntimeConfig.java]: Okay got it. Thanks.

**Comment by geoand** [comment]
Haven't worked much with Hibernate Search, but the Quarkus stuff looks good to me

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**